Piece

SunlayGGX opened the pull request.

Commit:
GameObjectManager a little modified

Not tested yet.

I added unspawnGO method and a different way to spawn GO

Commit:
Piece Class added

Not tested yet

I created a Piece object.

Commit:
Added a Piece Administrator

Not tested, not finished

I made a piece administrator, I've yet to test the thing. I haven't
finished though...

Commit:
Transform defult constructor changed

Tested. It works

I changed the default constructor because it introduced some bugs

Commit:
RendererManager cleaning + adding of some methods

Tested... And not tested

I added :
- removeAMesh method. It must be optimized and changed somehow because it isn't really clean but I don't have time for doing so
- init method. To init the RendererManager

inside the init, I create the dispositif. It won't be the moteur now...

Commit:
Cleaning of CMoteurWindow

Broken as it is


I remove the creationDispositifSpecific that creates the CDispositif now that it is the renderer manager that does the job...
I also removed the InitManipulator to move it to the input task init.
I made 2 getter to get specific windows instance handle

Commit:
Changed the size of the map

Tested and ok. But still broken due to the last commit

OK

Commit:
Fix the moteur and added new task

Still broken

- I removed the virtual method CreationDispositifSpecific.
- I added a new task named SpawnTask. Not here for now thus it is a broken commit due to the file will be missing

Commit:
I fixed the Input and render task

Still broken


I fixed the input task (made the init) and the renderTask (the same fix)

Commit:
Spawn Task implemented

Tested. OK


I made the spawnTask.

Commit:
Added Init and update to the GameObjectManager

Tested. OK, but not OK

I added init and an update of the PieceAdministrator inside the animAllGameObject.

Commit:
Fix bug with blocMesh.

Tested and works


A bug was inserted because of the stupid XMFLOAT3 default constructor...
I also added some way to use a custom PS and VS inside BlocMesh

Commit:
A lot of thing for Piece and PieceAdmin

Tested and it Works!!

I made the algorithm for the Piece and piece administrator. Now, it is OK...
Pieces are spawning (not tested the unspawn because I don't have collider box...)
They are yellow. They rotates, they ... Ok, everything fine.

Commit:
Removed a useless method pointer in Piece object

Tested and OK

I removed the method pointer because a if else was doing the job before
the uses of the anim of the piece. OK.

Commit:
Fix bug with GOManager.

Tested

PereKastor reviewed (changes requested):
J'aimerais que l'on parle de cette PR avant qu'elle puisse être mergé, quelques trucs ne vont pas dans ces changements et j'aimerais connaitre le pourquoi du comment :)

Commit:
Fix as Mr PereKastor wanted

Tested. Doesn't like it but works... somehow.

OK...

Commit:
Now. I think it is logic, lol

Tested. It works, buuut

I made Piece from GO. Really Huuuuuuuge fuck. I wasn't really okay with
remark Mr PereKastor and Mr ShabSendra made :
- I broke the GO logic...
- I separate the Piece from other GO (Piece were GO though but were
  ostracized from GO)
- and many other bad remarks.

I though what I were doing for Pieces before were good things but since these 2 made
some remarks about my way of thinking, OK, I changed it. It compiles and
works like before but It will pose really huge optimization problems.

My way of doing things before were way more optimized !!! This solution is
crappy but were imposed by 2 people... I cannot wait to see what the future
would be with this solution. But one thing for sure -> Not a solution to
do in a big game with a lot of GO...

PereKastor committed:
Fix timeManager bug, the elapsedTime is now really the elapsedTime

PereKastor committed:
Fix physicsTask order, it must be done before the renderTask.

Commit:
Redo the former design with PereKastor bug fix

Tested

Commit:
Merge branch 'Piece' of https://github.com/PirateTeamDDJV12/Pirate-Simulator into Piece

Commit:
Redo former design. End

Tested. OK
